src/postprocess/band_structure.jl

```diff
@@ -99,35 +83,37 @@ end
 end
 
 
-function split_into_branches(kcoords, data::Dict, klabels::Dict)
+function kdistances_and_ticks(kcoords, klabels::Dict)
     # kcoords in cartesian coordinates, klabels uses cartesian coordinates
     function getlabel(kcoord; tol=1e-4)
         findfirst(c -> norm(c - kcoord) < tol, klabels)
     end
 
-    branches = Any[(kindices = [0], kdistances=[0.0], ), ]
+    kdistances = eltype(kcoords[1])[]
+    tick_distances = eltype(kcoords[1])[]
+    tick_labels = String[]
     for (ik, kcoord) in enumerate(kcoords)
         previous_kcoord = ik == 1 ? kcoords[1] : kcoords[ik - 1]
-        if !isnothing(getlabel(kcoord)) && !isnothing(getlabel(previous_kcoord))
-            # New branch encountered
-            previous_distance = branches[end].kdistances[end]
-            push!(branches, (kindices=[ik], kdistances=[previous_distance]))
+        if ik == 1
+            # First k point
+            push!(kdistances, 0)
+            push!(tick_labels, getlabel(kcoord))
+            push!(tick_distances, kdistances[end])
+        elseif !isnothing(getlabel(kcoord)) && !isnothing(getlabel(previous_kcoord))
+            # New branch encountered. Do not increase the distance.
+            push!(kdistances, kdistances[end])
+            tick_labels[end] *= " | " * getlabel(kcoord)
         else
             # Keep adding to current branch
-            distance = branches[end].kdistances[end] + norm(kcoord - kcoords[ik - 1])
-            push!(branches[end].kdistances, distance)
-            push!(branches[end].kindices, ik)
+            distance = kdistances[end] + norm(kcoord - previous_kcoord)
+            push!(kdistances, distance)
+            if !isnothing(getlabel(kcoord))
+                push!(tick_labels, getlabel(kcoord))
+                push!(tick_distances, kdistances[end])
+            end
         end
     end
-
-    map(branches[2:end]) do branch
-        branch_data = Dict(key => data[key][branch.kindices, :, :] for key in keys(data))
-        ret = (klabels=(getlabel(kcoords[branch.kindices[1]]),
-                        getlabel(kcoords[branch.kindices[end]])),
-               kdistances=branch.kdistances,
-               kindices=branch.kindices)
-        merge(ret, (; branch_data...))
-    end
+    kdistances, tick_distances, tick_labels
```

Comment:
As a general comment: In such cases I would better return a named tuple compared to a tuple. You can still use it as a tuple, plus it has the additional benefit that the included "name" of the object carries more meaning.